test/test.js

```diff
@@ -1427,6 +1427,22 @@ describe('node-fetch', () => {
 		});
 	});
 
+	// issue #414
+	it('should reject if attempt to accumulate body stream throws', function () {
+		let body = resumer().queue('a=1').end();
+		body = body.pipe(new stream.PassThrough());
+		const res = new Response(body);
+		const bufferConcat = Buffer.concat;
+		const restoreBufferConcat = () => Buffer.concat = bufferConcat;
+		Buffer.concat = () => { throw new Error('embedded error'); };
+
+		return res.text().then(() => chai.assert(false)).catch(err => {
+			expect(err).to.be.an.instanceOf(FetchError)
+				.and.include({ type: 'system' })
+				.and.have.property('message').that.includes('Could not create Buffer')
+				.and.that.includes('embedded error');
+		}).then(restoreBufferConcat, restoreBufferConcat);
```

Comment:
Specifying these `.then()`s would mask the errors, since `restoreBufferConcat()` doesn't forward errors. This problem would be solved if we move to a Node.js version supporting [`promise.finally()`](https://github.com/tc39/proposal-promise-finally#why-not-thenf-f).

This should work:

```js
		const res = new Response(body);
		const bufferConcat = Buffer.concat;
		const restoreBufferConcat = () => Buffer.concat = bufferConcat;
		Buffer.concat = () => { throw new Error('embedded error'); };
 
		const textPromise = res.text();
		textPromise.then(restoreBufferConcat, restoreBufferConcat);
		return textPromise.then(() => chai.assert(false)).catch(err => {
			expect(err).to.be.an.instanceOf(FetchError)
				.and.include({ type: 'system' })
				.and.have.property('message').that.includes('Could not create Buffer')
				.and.that.includes('embedded error');
		});
```

Reply:
D'oh, silly oversight. Fixed. Thanks! (I did it slightly differently using `chai-as-promised`, but to essentially the same effect.)